proposals/nnnn-safe-shared-contiguous-storage.md

```diff
@@ -0,0 +1,1232 @@
+
+```swift
+extension MyResilientType {
+  // public API
+  @inlinable public func essentialFunction(_ a: some ContiguousStorage<some Any>) -> Int {
+    self.essentialFunction(a.storage)
+  }
+
+  // ABI boundary
+  public func essentialFunction(_ a: Span<some Any>) -> Int { ... }
+}
+```
+
+Here, the public function obtains the `Span` from the type that vends it in inlinable code, then calls a concrete, opaque function defined in terms of `Span`. Inlining the generic shim in the client is often a critical optimization. The need for such a pattern and related improvements are discussed in the future directions below (see [Syntactic Sugar for Automatic Conversions](#Conversions).)
+
+#### Extensions to Standard Library and Foundation types
+
+```swift
+extension Array: ContiguousStorage<Self.Element> {
+  // note: this could borrow a temporary copy of the `Array`'s storage
+  var storage: Span<Element> { get }
+}
+extension ArraySlice: ContiguousStorage<Self.Element> {
+  // note: this could borrow a temporary copy of the `ArraySlice`'s storage
+  var storage: Span<Element> { get }
+}
+extension ContiguousArray: ContiguousStorage<Self.Element> {
+  var storage: Span<Element> { get }
+}
+
+extension Foundation.Data: ContiguousStorage<UInt8> {
+  var storage: Span<UInt8> { get }
+}
+
+extension String.UTF8View: ContiguousStorage<Unicode.UTF8.CodeUnit> {
+  // note: this could borrow a temporary copy of the `String`'s storage
+  var storage: Span<Unicode.UTF8.CodeUnit> { get }
+}
+extension Substring.UTF8View: ContiguousStorage<Unicode.UTF8.CodeUnit> {
+  // note: this could borrow a temporary copy of the `Substring`'s storage
+  var storage: Span<Unicode.UTF8.CodeUnit> { get }
+}
+extension Character.UTF8View: ContiguousStorage<Unicode.UTF8.CodeUnit> {
+  // note: this could borrow a temporary copy of the `Character`'s storage
+  var storage: Span<Unicode.UTF8.CodeUnit> { get }
+}
+
+extension SIMD: ContiguousStorage<Self.Scalar> {
+  var storage: Span<Self.Scalar> { get }
+}
+extension KeyValuePairs: ContiguousStorage<(Self.Key, Self.Value)> {
+  var storage: Span<(Self.Key, Self.Value)> { get }
+}
+extension CollectionOfOne: ContiguousStorage<Element> {
+  var storage: Span<Element> { get }
+}
+
+extension Slice: ContiguousStorage where Base: ContiguousStorage {
+  var storage: Span<Base.Element> { get }
+}
+```
+
+In addition to the the safe types above gaining the `storage` property, the `UnsafeBufferPointer` family of types will also gain access to a `storage` property. This enables interoperability if `Span`-taking API. While a `Span` binding created from an `UnsafeBufferPointer` exists, the memory that underlies it must not be deinitialized or deallocated.
```

Comment:
This seems like a bad idea to me, because it encourages accidents. 

The right way to convert an UBP to a Span is to use one of the initializers that take an owner.

Reply:
I'll remove this. I've become less convinced myself

Reply:
Removed in favor of an example directly using a `Span` initializer.